### ISSUE TITLE
[bug] HCM: Fix styles for Segmented Button in High Contrast Mode (border width, hover state, and selected button styling)

### ISSUE THREAD

**Issue by alionazherdetska** [swisspost/design-system#4298]
**Description:**

**Border Width:**
The border width is inconsistent on devices with a width of less than 600px.
The border width on hover is incorrect and needs adjustment.

https://github.com/user-attachments/assets/96895e51-c366-41c3-b723-c5c26ac33cd9

**Selected Button:**
The selected button is missing a Selected background color.
The content inside the selected button is missing the SelectedItemText color
![Image](https://github.com/user-attachments/assets/dd0ceffe-b2d2-4759-b608-c0bdc4e1618c)
